lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -2442,11 +2449,16 @@ void TypeChecker::diagnoseIfDeprecated(SourceRange ReferenceRange,
   // We match the behavior of clang to not report deprecation warnings
   // inside declarations that are themselves deprecated on all deployment
   // targets.
-  if (Where.isDeprecated()) {
+  if (Where.isDeprecated() || Where.ignoresDeprecation()) {
     return;
   }
 
   auto *ReferenceDC = Where.getDeclContext();
+  auto *moduleImportedFrom = ReferenceDC->findImportFor(ReferenceDC->getParentModule());
```

Comment:
You'll want to query the module of the deprecated decl (`DeprecatedDecl->getDeclContext()->getParentModule()`), as `ReferenceDC->getParentModule()` is the module of the reference.

Also you'll want to remove the `*` as `moduleImportedFrom` is not a pointer. IMO something like `declImport` or just `import` would be a better name for the variable as it's the import, not the module, what do you think?

Reply:
Sure, i'll rename & remove the `*`